### PR TITLE
Fix for the issue #378

### PR DESCRIPTION
**Pull request by SivaMaplelabs**

### CODE REVIEW

File: bin/upload-data.py [hunk 1]
```diff
@@ -69,8 +71,14 @@ def main():
     if not os.path.isabs(folder):
         folder = os.path.join(config.service_folder, folder)
         print("Folder is relative, using", folder)
+    files += args.files
     if args.common is not None:
-        copy_everywhere(config, args.common, folder, copyOptions)
+        files.append(args.common)
```
Comment (mbudiu-vmw) [inline bin/upload-data.py]: Only args.common file is copied to all machines, the other files are copied in round-robin amongst all machines.

Comment (SivaMaplelabs) [reply bin/upload-data.py]: Yeah,  got it.  Will change.

Comment (SivaMaplelabs) [reply bin/upload-data.py]: @mbudiu-vmw Done.

File: bin/upload-data.py [hunk 2]
```diff
@@ -71,6 +71,11 @@ def main():
         print("Folder is relative, using", folder)
     if args.common is not None:
         copy_everywhere(config, args.common, folder, copyOptions)
+    if args.files:
+        copy_files(config, folder, args.files, copyOptions)
+    else:
+        print("No files to upload to the machines provided in a Hillview configuration")
+        exit(1)
```
Comment (mbudiu-vmw) [inline bin/upload-data.py]: There is no need to exit with 1 - sometimes you only want to use the args.common option

**Comment by mbudiu-vmw** [comment]
I also don't see how this fixes issue #378 
We should check whether that issue is indeed correct.
I don't know when rsync decides to update a file.

File: bin/upload-data.py (hunk 2 above)
Comment (SivaMaplelabs) [reply bin/upload-data.py]: @mbudiu-vmw Done.

**Comment by SivaMaplelabs** [comment]
Currently, this script is not uploading the files to the given machines in a round-robin fashion. Only the schema file is uploading to all the machines.

**Comment by SivaMaplelabs** [comment]
Only copying the schema, not the files.
https://github.com/vmware/hillview/blob/master/bin/upload-data.py#L72

**Pull request merged**